pkg/operator/observe_config.go

```diff
@@ -97,6 +101,58 @@ func (c ConfigObserver) sync() error {
 	return nil
 }
 
+// observeClusterConfig observes cloud provider configuration from
+// cluster-config-v1 in order to configure kube-controller-manager's cloud
+// provider.
+func observeClusterConfig(kubeClient kubernetes.Interface, clientConfig *rest.Config, observedConfig map[string]interface{}) (map[string]interface{}, error) {
+	clusterConfig, err := kubeClient.CoreV1().ConfigMaps("kube-system").Get("cluster-config-v1", metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		glog.Warningf("cluster-config-v1 not found in the kube-system namespace")
+		return observedConfig, nil
+	}
+	if err != nil {
+		return observedConfig, err
+	}
+
+	installConfigYaml, ok := clusterConfig.Data["install-config"]
+	if !ok {
+		return observedConfig, nil
+	}
+	installConfig := map[string]interface{}{}
+	err = yaml.Unmarshal([]byte(installConfigYaml), &installConfig)
+	if err != nil {
+		glog.Warningf("Unable to parse install-config: %s", err)
+		return observedConfig, nil
+	}
+
+	// extract needed values
+	//  data:
+	//   install-config:
+	//     platform:
+	//       aws: {}
+	cloudProvider := ""
+	platform, ok := installConfig["platform"].(map[string]interface{})
+	if !ok {
```

Comment:
check for empty values as well

Reply:
Does it matter given we check for unmarshaling errors?

Reply:
Oops, previous comment applies to another line... what do you mean re: empty value here? if the type conversion works I have a map to work with